diff --git a/packages/artplayer-plugin-jassub/src/jassub.es.js b/packages/artplayer-plugin-jassub/src/jassub.es.js
index 904b40888..6bd2bc2c6 100644
--- a/packages/artplayer-plugin-jassub/src/jassub.es.js
+++ b/packages/artplayer-plugin-jassub/src/jassub.es.js
@@ -143,7 +143,7 @@ class o extends EventTarget {
         // @ts-ignore
         useLocalFonts: typeof queryLocalFonts < "u" && (e.useLocalFonts ?? !0),
         hasBitmapBug: o._hasBitmapBug
-      }), this._offscreenRender === !0 && this.sendMessage("offscreenCanvas", null, [this._canvasctrl]);
+      });
     }).catch((error) => {
       if (!this._destroyed) this.destroy(error);
     });
@@ -580,6 +580,10 @@ class o extends EventTarget {
   }
   _ready() {
     if (this._destroyed) return;
+    if (this._offscreenRender === !0 && !this._initialOffscreenSent) {
+      this._initialOffscreenSent = true;
+      this._worker.postMessage({ target: "offscreenCanvas", transferable: [this._canvasctrl] }, [this._canvasctrl]);
+    }
     this._init(), this.dispatchEvent(new CustomEvent("ready"));
   }
   /**
